mux_test.go

```diff
@@ -1014,6 +1004,26 @@ func TestBuildVarsFunc(t *testing.T) {
 	}
 }
 
+// Addresses https://github.com/gorilla/mux/issues/200
+func TestCorrectVars(t *testing.T) {
+	tests := []routeTest{
+		{
+			title:        "Correct nested regex vars",
+			route:        new(Route).PathPrefix("/{id}/{kind:(a|b)}/{idx:[0-9]+}"),
```

Comment:
Yes - most users probably don't know the difference between a capturing and non-capturing group (or care), we could show an error.

However, the only path to doing so is a panic, and we'd effectively have to use a regex on the pattern first to look for non-capturing groups...